lib/Solver/FastCexSolver.cpp

```diff
@@ -269,13 +270,15 @@ class ValueRange {
 
     std::uint64_t smallest = (static_cast<std::uint64_t>(1) << (bits - 1));
     if (m_max >= smallest) {
-      return ints::sext(smallest, 64, bits);
+      auto sign_extended = smallest | -smallest;
+      return toSigned(sign_extended);
     } else {
       return m_min;
     }
   }
 
   std::int64_t maxSigned(unsigned bits) const {
+    assert(bits >= 2 && bits <= 64);
```

Comment:
Can we do an APInt::getSignedMaxValue(bits).getSExtValue() instead ?

Reply:
No, but we can use `llvm::APInt(m_max, bits, true).getSExtValue()`

Reply:
Implemented